### PR TITLE
use reusable workflows

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
This PR modify the workflows in order to use the ones stored in `qiboteam/workflows` (see qiboteam/workflows#23).

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[tool.pylint.master]
+exit-zero = true
+errors-only = true
```
Comment (AleCandido) [inline pyproject.toml:3]:
```suggestion
```

**Review by AleCandido — CHANGES_REQUESTED** [review]

**Comment by Edoardo-Pedicillo** [comment]
@scarrazza This PR is ready to be merged.

**Review by AleCandido — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 [tool.pylint.master]
+exit-zero = true
```
Comment (AleCandido) [inline pyproject.toml]: Wait a second: you should not always exit with zero!

The first check on errors should actually crash if failing.

File: pyproject.toml [hunk 3]
```diff
@@ -1,4 +1,6 @@
...
+errors-only = true
```
Comment (AleCandido) [inline pyproject.toml]: And you should also check for warnings, why did you revert?

**Review by AleCandido — CHANGES_REQUESTED** [review]

**Review by AleCandido — APPROVED** [review]
My bad, I was only watching at one commit, but not the one after (current last)...

**Review by scarrazza — APPROVED** [review]

**Pull request merged**